### PR TITLE
added mongo tracker store

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- added mongo tracker store
- made sure the redis tracker store calls the event endpoint

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_core/tracker_store.py [hunk 1]
```diff
@@ -136,3 +139,58 @@ def retrieve(self, sender_id):
...
+        self.db = Database(self.client, db)
+        self.collection = collection
+        super(MongoTrackerStore, self).__init__(domain, event_broker)
+
+        self._ensure_indices()
+
+    def _conversations(self):
+        return self.db[self.collection]
+
+    def _ensure_indices(self):
```
Comment (ricwo) [inline rasa_core/tracker_store.py]: `_ensure_index()` as it's only one?

File: rasa_core/tracker_store.py [hunk 2]
```diff
@@ -136,3 +139,58 @@ def retrieve(self, sender_id):
...
+                                  # delay connect until process forking is done
+                                  connect=False)
+
+        self.db = Database(self.client, db)
+        self.collection = collection
+        super(MongoTrackerStore, self).__init__(domain, event_broker)
+
+        self._ensure_indices()
+
+    def _conversations(self):
```
Comment (ricwo) [inline rasa_core/tracker_store.py]: should we make this a `@property`?

File: rasa_core/tracker_store.py [hunk 3]
```diff
@@ -136,3 +139,64 @@ def retrieve(self, sender_id):
...
+
+        self._conversations().update_one(
+                {"sender_id": tracker.sender_id},
+                {"$set": state},
+                upsert=True)
+
+    def retrieve(self, sender_id):
+        stored = self._conversations().find_one({"sender_id": sender_id})
+        if stored is not None:
+            if self.domain:
```
Comment (ricwo) [inline rasa_core/tracker_store.py]: 👍

File: rasa_core/tracker_store.py (hunk 1 above)
Comment (tmbo) [reply rasa_core/tracker_store.py]: it's called like this everywhere else, lets keep the plural - maybe we want to add another index in the future :D .

**Pull request merged**